### PR TITLE
Update Core dependency to 10.0.0-beta.21

### PR DESCRIPTION
**Pull request by nishant-karajgikar**
PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-ios/blob/main/CONTRIBUTING.md#contributor-license-agreement).

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Document any changes to public APIs.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'
 - [x] Add an entry inside this element for inclusion in the `mapbox-maps-ios` changelog: `<changelog>Update to MapboxCoreMaps 10.0.0-beta.21</changelog>`.

### Summary of changes

This PR updates our dependency to MapboxCoreMaps 10.0.0-beta.21. It results in the following changes:

- Move to using `CameraState` vs. `CameraOptions` as appropriate.
- Update ResourceOptions to rely on `MBMTileStoreUsageMode`
- Rename `BoundOptions` --> `CameraBoundsOptions`

### CODE REVIEW

File: Tests/MapboxMapsTests/Foundation/Camera/MapboxMapsCameraTests.swift [hunk 1]
```diff
@@ -1,78 +1,78 @@
...
-        })
-
-        let expectedCamera = CameraOptions(pitch: 86)
-
-        cameraManager.setCamera(to: expectedCamera)
-
-        XCTAssertEqual(mapView.pitch, cameraManager.mapCameraOptions.maximumPitch, accuracy: 0.000001)
-    }
-}
+//internal class CameraManagerIntegrationTests: MapViewIntegrationTestCase {
```
Comment (nishant-karajgikar) [inline Tests/MapboxMapsTests/Foundation/Camera/MapboxMapsCameraTests.swift:4]: These tests are flakey and unnecessary.. We'll soon be moving to an approach where we will rely on the clamping behavior that Gl-native provides. I'll get rid of this file at that point.

File: Apps/StressTest/StressTest/ViewController.swift [hunk 2]
```diff
@@ -290,7 +290,7 @@ class ViewController: UIViewController {
         print("Creating snapshotter")
         let snapshotter = Snapshotter(options: options)
         snapshotter.style.uri = .light
-        snapshotter.camera = mapView.cameraOptions
+        snapshotter.setCamera(to: CameraOptions(cameraState: mapView.cameraState))
```
Comment (julianrex) [inline Apps/StressTest/StressTest/ViewController.swift:293]: I wonder if we're going to need a shorthand for this.

File: Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift [hunk 3]
```diff
@@ -12,15 +12,15 @@ extension GestureManager: GestureHandlerDelegate {
...
+            cameraManager.setCamera(to: CameraOptions(zoom: CGFloat(mapView.cameraState.zoom + 1.0)),
                                     animated: true,
                                     duration: 0.3,
                                     completion: nil)
         }
 
         // Double tapping twice with two fingers will cause the map to zoom out
         if numberOfTaps == 2 && numberOfTouches == 2 {
-            cameraManager.setCamera(to: CameraOptions(zoom: mapView.zoom - 1.0),
+            cameraManager.setCamera(to: CameraOptions(zoom: CGFloat(mapView.cameraState.zoom - 1.0)),
```
Comment (julianrex) [inline Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift:23]: Where's the mismatch here?

**Review by julianrex — APPROVED** [review]
Looks good - thanks!

File: Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift (hunk 3 above)
Comment (nishant-karajgikar) [reply Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift]: ah right.. this was an oversight. Addressed now.

File: Apps/StressTest/StressTest/ViewController.swift (hunk 2 above)
Comment (nishant-karajgikar) [reply Apps/StressTest/StressTest/ViewController.swift:293]: Potentially.. We do need to do some communication around `CameraState` vs `CameraOptions`.

**Pull request merged**